### DISCUSSION TITLE
next/image "priority" prop results in fetchpriority="high" on <img>

### DISCUSSION THREAD

**Discussion by kevinpfox** [vercel/next.js#39740] [Ideas]
Currently if you do...
`<Image priority ...>`
... it will prevent that `<img>` from lazy loading.
How about having Next code add `fetchpriority="high"` to img tag too?
https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/fetchPriority
https://web.dev/optimize-lcp/#optimize-when-the-resource-is-discovered

When I first saw that next/image priority, I was like Eureka! It must add `fetchpriority="high"` to `<img>` and then I was surprised to discover it doesn't do that.

It would be great if `priority` took the same string values as HTMLImageElement.fetchPriority
'high', 'low', 'auto'

It could default to 'auto' + lazy loading but could overridden to 'high' (no lazy load and fetchpriority attribute added)

**Comment by kevinpfox** [comment]
I just discovered that you can do....
`<Image priority fetchpriority='high' ...>`
... and it passes fetchpriority down to the `<img>` tag and I get the result I was looking for. Cool.

**Reply by GRASPARK** [reply]
Hello @kevinpfox,
On which version of Next JS did you try that as with Next JS Image Component none of Props are currently working with ***Fetch Priority*** value..?

**Reply by kevinpfox** [reply]
Our current version: `next:12.3.0`

For ref:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attr-fetchpriority

It's still not fully supported, perhaps why Next.js team isn't going to "going there" as of yet?

![ss2](https://user-images.githubusercontent.com/57231794/216395247-f037933f-1009-40b9-8c8c-2e457802738b.png)


![ss0](https://user-images.githubusercontent.com/57231794/216394229-39cdfda1-ccf5-4084-82c7-02dc50d6d262.png)

![ss1](https://user-images.githubusercontent.com/57231794/216394239-1f41ecbd-dfd8-4cee-a595-73aa5e75006a.png)

**Comment by kevinpfox** [comment]
For LCP, It would be cool if `Image` `priority` prop added `fetchpriority="high"` to the the link preload tag that next/image adds to the top of the document. That is a recommended according to https://web.dev/optimize-lcp/#optimize-when-the-resource-is-discovered

`<link rel="preload" fetchpriority="high" as="image" href="/path/to/hero-image.jpg">`

**Reply by mpfullstack** [reply]
Hi Kevin,

I'm trying this but I do not see fetchpriority being added to <link ... preload meta tag. 
What version of NextJS is it working this for you?

Thank you very much

**Reply by hckhanh** [reply]
> For LCP, It would be cool if `Image` `priority` prop added `fetchpriority="high"` to the the link preload tag that next/image adds to the top of the document. That is a recommended according to https://web.dev/optimize-lcp/#optimize-when-the-resource-is-discovered
> 
> `<link rel="preload" fetchpriority="high" as="image" href="/path/to/hero-image.jpg">`

+1

**Comment by erakor** [comment]
Hi, re-upping this discussion.

Are there any plans/appetite to make the `priority` prop add the `fetchpriority` attribute to the image and/or the preload tag?

**Comment by styfle** [comment]
This was released in Next.js 13.3.0, thanks!

https://github.com/vercel/next.js/releases/tag/v13.3.0
